### PR TITLE
Get rid of breakpoint_table, fix crash

### PR DESCRIPTION
**Pull request by Keno**
In #2473, I remarked that rr segfaults when attempting to increase
the size of the syscall buffer. That turned out to be because the
breakpoint_table hardcodes the syscall buffer size as written.
Rather than attempting to dynamically size the breakpoint table
(and deal with all the ugliness, like dynamically registering
unwind info for it), this instead replaces the breakpoint_table
entirely. Rather than relying on a debugger trap to mark the end
of the replayable syscall buffer, we simply inform the preload
library of the byte offset we would like it to stop at and then
induce a deterministic SIGSEGV during replay once this value has
been reached. Of course, we must be careful not to use any
conditional branches in this endavor, but doing so is not too
difficult. The result is a mechanism that does not care about
the size of the allocated syscalbuffer and moreover saves us
some memory in every task during replay.

The only bit of ugliness here is that we do want to retain the
ability to replay old traces, so we retain some code to support
that even though it won't be used for any traces generated from
this point forward - perhaps we can delete it if there is ever
an incompatible break in the trace format in the future.

### CODE REVIEW

**Comment by Keno** [comment]
Once sec, looks like I didn't quite get the .S excision right, so it doesn't built from a fresh checkout.

**Comment by Keno** [comment]
Fixed.

File: src/preload/syscallbuf.c [hunk 1]
```diff
@@ -884,6 +884,29 @@ static int start_commit_buffered_syscall(int syscallno, void* record_end,
   return 1;
 }
 
+static void do_breakpoint(size_t value)
+{
+  char *unsafe_value = ((char*)-1)-0xf;
+  char *safe_value = (char*)&unsafe_value;
```
Comment (rocallahan) [inline src/preload/syscallbuf.c]: Why not just use null for `unsafe_value`? And why not make `safe_value` a `char**` to avoid the cast?

**Comment by rocallahan** [comment]
> perhaps we can delete it if there is ever an incompatible break in the trace format in the future.

I don't ever intend to make such a break.

File: src/preload/syscallbuf.c (hunk 1 above)
Comment (Keno) [reply src/preload/syscallbuf.c]: NULL may be dereferencable if set up via personality(2) (though frankly the only use I know of is for exploits). As far as I know the value I chose here should always cause a fault.

**Comment by rocallahan** [comment]
Thanks.

**Pull request merged**